webapp/channels/src/components/setting_item.tsx

```diff
@@ -41,37 +41,42 @@ type Props = {
     collapsedEditButtonWhenDisabled?: ReactNode;
 }
 
-export default class SettingItem extends React.PureComponent<Props> {
-    minRef: RefObject<SettingItemMinComponent>;
-
-    constructor(props: Props) {
-        super(props);
-
-        this.minRef = React.createRef();
-    }
+const SettingItem = ({
+    active,
+    areAllSectionsInactive,
+    section,
+    max,
+    updateSection,
+    title,
+    isDisabled,
+    describe,
+    collapsedEditButtonWhenDisabled,
+}: Props) => {
+    const minRef = useRef<SettingItemMinComponent>(null);
 
-    componentDidUpdate(prevProps: Props) {
+    useEffect(() => {
         // We want to bring back focus to the edit button when the section is opened and then closed along with all sections are closed
-        if (!this.props.active && prevProps.active && this.props.areAllSectionsInactive) {
-            this.minRef.current?.focus();
-        }
-    }
 
-    render() {
-        if (this.props.active) {
-            return this.props.max;
+        if (!active && areAllSectionsInactive) {
+            minRef.current?.focus();
         }
+    }, [active, areAllSectionsInactive]);
```

Comment:
In order to keep the exact behavior it had before, we should remove the `areAllSectionsInactive` from the dependency list.

See that on the componentDidUpdate, we only execute when active changes (`!this.props.active && prevProps.active`). Changes in `areAllSectionsInactive` should not trigger the focus (at least according to the original implementation).

```suggestion
    }, [active]);
```

Reply:
yes, I will update the dependency list of the useEffect so it is only triggered by the  `active` prop change